sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -172,18 +171,19 @@ public ValidationResult validate(boolean authorizeContextParams) throws SqlParse
    * Prepare an SQL query for execution, including some initial parsing and validation and any dynamic parameter type
    * resolution, to support prepared statements via JDBC.
    *
-   * In some future this could perhaps re-use some of the work done by {@link #validate(boolean)}
+   * In some future this could perhaps re-use some work done by {@link #validate(boolean)}
    * instead of repeating it, but that day is not today.
    */
+  // RelConversionException is no longer thrown, but removing it causes
```

Comment:
nit: is it really that many files?

Reply:
Yes! It expands ever-outward. Is good to clean up, but as a separate PR to keep this one small.

Reply:
Right. From SqlLifecycle, the exception is exposed to the callers, which include the tests. Many tests then expose the exception in their test function signatures. So, to fully clean up, we have to clean up many test functions. That's the work to be done in another PR.